File: src/main/java/ru/reosfire/wmlib/text/Replacement.java
package ru.reosfire.wmlib.text;

import java.util.LinkedList;

public class Replacement
{
    private final String from;
    private final String to;

    public Replacement(String from, String to)
    {
        this.from = from;
        this.to = to;
    }

    public String set(String Input)
    {
        if (Input == null) return null;
        if (from == null) return Input;
        if (to == null) return Input;
        return Input.replace(from, to);
    }

    public static String set(String message, Replacement replacement)
    {
        return replacement.set(message);
    }

    public static String set(String message, Replacement... replacements)
    {
        if (replacements == null) return message;
        for (Replacement replacement : replacements)
        {
            message = replacement.set(message);
        }
        return message;
    }

    public static Iterable<String> set(Iterable<String> messages, Replacement replacement)
    {
        LinkedList<String> result = new LinkedList<>();
        for (String message : messages)
        {
            result.add(replacement.set(message));
        }
        return result;
    }

    public static Iterable<String> set(Iterable<String> messages, Replacement... replacements)
    {
        if (replacements == null) return messages;
        Iterable<String> temp = messages;
        for (Replacement replacement : replacements)
        {
            temp = set(messages, replacement);
        }
        return temp;
    }
}

File: src/main/java/ru/reosfire/wmlib/text/Replacement.kt
package ru.reosfire.wmlib.text

import java.util.*

class Replacement(private val from: String, private val to: String) {
    fun set(Input: String): String {
        return Input.replace(from, to)
    }

    companion object {
        operator fun set(message: String, replacement: Replacement): String {
            return replacement.set(message)
        }

        fun set(message: String, vararg replacements: Replacement): String {
            var result = message
            for (replacement in replacements) {
                result = replacement.set(message)
            }
            return result
        }

        operator fun set(messages: Iterable<String>, replacement: Replacement): Iterable<String> {
            val result = LinkedList<String>()
            for (message in messages) {
                result.add(replacement.set(message))
            }
            return result
        }

        fun set(messages: Iterable<String>, vararg replacements: Replacement): Iterable<String> {
            var temp = messages
            for (replacement in replacements) {
                temp = set(messages, replacement)
            }
            return temp
        }
    }
}